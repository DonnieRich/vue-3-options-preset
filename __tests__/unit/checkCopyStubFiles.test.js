const { vol } = require("memfs");
jest.mock("fs/promises");

const { copyStubFiles } = require("../../src/bin/copyStubFiles");

// import all json config files
const { config } = require('../../src/config/config');
const { BASE_DIR, BASE_STUBS_DIR } = config.get();

describe(copyStubFiles, () => {
    beforeEach(() => {
        vol.reset();
    });

    it("Should pass if the stub files are copied in the new location", async () => {

        vol.fromJSON(
            {
                [`${BASE_STUBS_DIR}/default/App.vue`]: "New App.vue",
                [`${BASE_STUBS_DIR}/default/general.scss`]: "General SCSS",
                [`${BASE_STUBS_DIR}/default/HelloWorld.vue`]: "New component",
                [`${BASE_STUBS_DIR}/default/main.js`]: "New main.js",
                [`${BASE_DIR}/justAFile.js`]: "...",
                [`${BASE_DIR}/components/HelloWorld.vue`]: "Old component",
                [`${BASE_DIR}/App.vue`]: "Old App.vue"
            },
            "/"
        );

        await copyStubFiles(`${BASE_STUBS_DIR}/default`, BASE_DIR);
        expect(vol.toJSON()).toMatchSnapshot();
    });

    it("Should pass if the stub files are copied in the new location with SCSS file for Bootstrap", async () => {

        vol.fromJSON(
            {
                [`${BASE_STUBS_DIR}/bootstrap/App.vue`]: "New App.vue WITH BOOTSTRAP",
                [`${BASE_STUBS_DIR}/bootstrap/general.scss`]: "General SCSS WITH BOOTSTRAP",
                [`${BASE_STUBS_DIR}/bootstrap/HelloWorld.vue`]: "New component",
                [`${BASE_STUBS_DIR}/bootstrap/main.js`]: "New main.js WITH BOOTSTRAP",
                [`${BASE_DIR}/justAFile.js`]: "...",
                [`${BASE_DIR}/components/HelloWorld.vue`]: "Old component",
                [`${BASE_DIR}/App.vue`]: "Old App.vue"
            },
            "/"
        );

        await copyStubFiles(`${BASE_STUBS_DIR}/bootstrap`, BASE_DIR);
        expect(vol.toJSON()).toMatchSnapshot();
    });

    it("Should fail if the stub files for Bootstrap are missing", async () => {

        vol.fromJSON(
            {
                [`${BASE_STUBS_DIR}/default/App.vue`]: "New App.vue",
                [`${BASE_STUBS_DIR}/default/general.scss`]: "General SCSS",
                [`${BASE_STUBS_DIR}/default/HelloWorld.vue`]: "New component",
                [`${BASE_STUBS_DIR}/default/main.js`]: "New main.js",
                [`${BASE_DIR}/justAFile.js`]: "...",
                [`${BASE_DIR}/components/HelloWorld.vue`]: "old component",
                [`${BASE_DIR}/App.vue`]: "Old App.vue"
            },
            "/"
        );

        await expect(() => copyStubFiles(`${BASE_STUBS_DIR}/bootstrap`, BASE_DIR)).rejects.toThrow(Error);
        expect(vol.toJSON()).toMatchSnapshot();
    });

    it("Should fail if the stub files are missing", async () => {

        vol.fromJSON(
            {
                [`${BASE_STUBS_DIR}/bootstrap/App.vue`]: "New App.vue WITH BOOTSTRAP",
                [`${BASE_STUBS_DIR}/bootstrap/general.scss`]: "General SCSS WITH BOOTSTRAP",
                [`${BASE_STUBS_DIR}/bootstrap/main.js`]: "New main.js WITH BOOTSTRAP",
                [`${BASE_DIR}/justAFile.js`]: "...",
                [`${BASE_DIR}/components/HelloWorld.vue`]: "old component",
                [`${BASE_DIR}/App.vue`]: "Old App.vue"
            },
            "/"
        );

        await expect(() => copyStubFiles(`${BASE_STUBS_DIR}/default`, BASE_DIR)).rejects.toThrow(Error);
        expect(vol.toJSON()).toMatchSnapshot();
    });
});