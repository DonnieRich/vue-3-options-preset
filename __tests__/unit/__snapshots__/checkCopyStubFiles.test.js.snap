// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`copyStubFiles Should fail if the stub files are missing 1`] = `
{
  "/node_modules/vue-3-options-preset/src/stubs/bootstrap/App.vue": "New App.vue WITH BOOTSTRAP",
  "/node_modules/vue-3-options-preset/src/stubs/bootstrap/general.scss": "General SCSS WITH BOOTSTRAP",
  "/node_modules/vue-3-options-preset/src/stubs/bootstrap/main.js": "New main.js WITH BOOTSTRAP",
  "/src/App.vue": "Old App.vue",
  "/src/components/HelloWorld.vue": "old component",
  "/src/justAFile.js": "...",
}
`;

exports[`copyStubFiles Should fail if the stub files for Bootstrap are missing 1`] = `
{
  "/node_modules/vue-3-options-preset/src/stubs/default/App.vue": "New App.vue",
  "/node_modules/vue-3-options-preset/src/stubs/default/HelloWorld.vue": "New component",
  "/node_modules/vue-3-options-preset/src/stubs/default/general.scss": "General SCSS",
  "/node_modules/vue-3-options-preset/src/stubs/default/main.js": "New main.js",
  "/src/App.vue": "Old App.vue",
  "/src/components/HelloWorld.vue": "old component",
  "/src/justAFile.js": "...",
}
`;

exports[`copyStubFiles Should pass if the stub files are copied in the new location 1`] = `
{
  "/node_modules/vue-3-options-preset/src/stubs/default/App.vue": "New App.vue",
  "/node_modules/vue-3-options-preset/src/stubs/default/HelloWorld.vue": "New component",
  "/node_modules/vue-3-options-preset/src/stubs/default/general.scss": "General SCSS",
  "/node_modules/vue-3-options-preset/src/stubs/default/main.js": "New main.js",
  "/src/App.vue": "New App.vue",
  "/src/components/HelloWorld.vue": "New component",
  "/src/justAFile.js": "...",
  "/src/main.js": "New main.js",
  "/src/styles/general.scss": "General SCSS",
}
`;

exports[`copyStubFiles Should pass if the stub files are copied in the new location with SCSS file for Bootstrap 1`] = `
{
  "/node_modules/vue-3-options-preset/src/stubs/bootstrap/App.vue": "New App.vue WITH BOOTSTRAP",
  "/node_modules/vue-3-options-preset/src/stubs/bootstrap/HelloWorld.vue": "New component",
  "/node_modules/vue-3-options-preset/src/stubs/bootstrap/general.scss": "General SCSS WITH BOOTSTRAP",
  "/node_modules/vue-3-options-preset/src/stubs/bootstrap/main.js": "New main.js WITH BOOTSTRAP",
  "/src/App.vue": "New App.vue WITH BOOTSTRAP",
  "/src/components/HelloWorld.vue": "New component",
  "/src/justAFile.js": "...",
  "/src/main.js": "New main.js WITH BOOTSTRAP",
  "/src/styles/general.scss": "General SCSS WITH BOOTSTRAP",
}
`;
